### PR TITLE
[3.x] Tilemap editor - prevent changing tool when mouse buttons pressed

### LINKED ISSUE #85776: Godot crashes when using the tilemap node picker tool.
### Tested versions

v3.6.beta3.official [21ab700f2]
_bugsquad edit_ 3.5.2 stable

### System information

w10 64

### Issue description

Simply use the picker tool from tilemap mode indiscriminately ( use the shortcut key I ).
After a few actions the editor will have crashed and will not be able to assign values to the nodes or save the scene, among other things.

Simply use the picker tool ( use the shortcut key I ) many times as you can see in the video.
You will see that after several uses of the picker tool, I can now no longer assign a value to the sprite node.

If you continue doing this, the editor may close suddenly.

https://github.com/godotengine/godot/assets/19652075/3184bf7c-8ffd-40dc-b994-9ab699a14812

### Steps to reproduce

Simply use the picker tool from tilemap mode indiscriminately ( use the shortcut key I ).

### Minimal reproduction project (MRP)

[tilemapCrash.zip](https://github.com/godotengine/godot/files/13578410/tilemapCrash.zip)

### PR DESCRIPTION
**Pull request by lawnjelly**
Changing tool when painting prevented the corresponding commit of undo action when the mouse button was released. This led to undo actions getting out of sync and the undo system breaking the editor.

This PR simply prevents changing tool while mouse buttons are pressed, and prevents the above scenario.

Fixes #85776

## Discussion
This bug is quite nasty and exposes (imo) a problem with the undo system paradigm, when used in combination with mouse drawing as it is in the tile map editor plugin.

The problem afaik is that the undo system is a state machine, and undo actions are created, and then committed at a later time. If there is a client side bug (as here) the undo action is created but never committed, and the undo stack gets full, and the whole system breaks.

Ideally each create action would be matched and verified with the commit (via e.g. an ID), however there is a lot of existing code that relies on the current undo system as is.

Here I've taken one of the simplest possible courses to fix the bug locally, by preventing the mechanism which gets the undo action out of sync - preventing changing tool while drawing.

## Notes
* There are several other possible ways of addressing this, including committing the current draw when the tool is selected, which I tried first, but is considerably more complex to implement for a "fringe case".
* Ideally the undo system would feature detection code to prevent this kind of thing happening, or flag it up, as it likely occurs in other editor areas other than just the tilemap plugin.
* There is the possibility here of the local mouse button state getting out of sync (e.g. releasing mouse outside a window) so this would need thorough testing in case of regressions.
* This bug has likely existed for years without being reported afaik (likely users would just reboot the editor), so it is not a _must have_ for 3.6 imo.

### CODE REVIEW

**Comment by lawnjelly** [comment]
I can only give vague recommendations for merge timing, the 3.x team can discuss this.

We are hoping to be at feature freeze at the next beta (which is actually overdue), and while this isn't a feature, it is a good idea to note on PRs where something requires a bit of testing in the field, as it has the potential to require editor reboots if there is a bug (and input is kind of hard to predict). It might be that the RCs would be enough to bug test this.

Your desire for this is noted of course. :+1: 

Before anything else, it requires approval from a maintainer (and it might be decided that a more thorough fix is needed for the undo system, which would much more likely require deferring to 3.7).

**Review by groud — APPROVED** [review]
Sounds like an acceptable fix to me.

**Comment by akien-mga** [comment]
Thanks!

**Comment by Torguen** [comment]
Great, this seems to work correctly and there are no crashes, thanks!

**Pull request merged**
